sympy/functions/elementary/exponential.py

```diff
@@ -1173,6 +1173,14 @@ def _eval_is_algebraic(self):
         else:
             return s.is_algebraic
 
+    def _eval_as_leading_term(self, x, cdir=0):
+        if len(self.args) == 1:
+            arg0 = self.args[0]
+            arg = self.args[0].subs(x, 0).cancel()
```

Comment:
```suggestion
            arg = self.args[0]
            arg0 = arg.subs(x, 0).cancel()
```
Also `arg0` and `arg` swapped on the next lines.